desktop-exporter/app/components/traceList.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from "react";
+import { FixedSizeList } from "react-window";
+import { NavLink, useLoaderData } from "react-router-dom";
+import { Flex } from "@chakra-ui/react";
+import { useSize } from "@chakra-ui/react-use-size";
+
+import { TraceSummaries, TraceSummary } from "../types/api-types";
+
+type RowProps = {
+  index: number;
+  style: Object;
+  data: TraceSummary[];
+};
+
+function Row({ index, style, data }: RowProps) {
+  return (
+    <NavLink
+      to={`traces/${data[index].traceID}`}
+      style={style}
+    >
+      {data[index].traceID}
+    </NavLink>
+  );
+}
+
+export function TraceList() {
+  const { traceSummaries } = useLoaderData() as TraceSummaries;
```

Comment:
This does work fine, but I think I would recommend requiring this data higher up in the hierarchy and passing it down to `TraceList` as a `prop`.

Ideally the component directly rendered by the router, `MainView` would handle fetching the data from its `loader` so that `TraceList` is more self-contained and doesn't have dependencies on something several levels above itself.

In this way the coupling between `mainLoader` and `MainView` that we can't property type is constrained to one file / component.

```jsx
export default function MainView() {
  let [isFullWidth, setFullWidth] = useBoolean();
  let { traceSummaries } = useLoaderData() as TraceSummaries;
  
  return (
    <div className="container">
      <Sidebar
        isFullWidth={isFullWidth}
        toggle={setFullWidth.toggle}
        traceSummaries={traceSummaries}
      />
      <Outlet />
    </div>
  );
}
```